salt/_states/docker_registry.py

```diff
@@ -0,0 +1,202 @@
+'''
+States to manage images in a Docker registry.
+'''
+
+import logging
+
+import salt.exceptions
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'docker_registry'
+
+
+def __virtual__():
+    if 'docker_registry.get_image_manifest' not in __salt__:
+        return (False, "Missing 'docker_registry' module")
+
+    return __virtualname__
+
+
+def image_managed(name, archive_path=None, digest=None, force=False,
+        tls_verify=True):
+    '''
+    Push an image to a Docker repository.
+
+    name
+        Name of the image (including repository), e.g.,
+        `localhost:5000/metalk8s/nginx:1.14.2`.
+    archive_path : None
+        Path of the Docker archive to upload, if image not available.
+    digest : None
+        Expected digest of the image. If no match and `force` is `False`, the
+        state will fail.
+    force : False
+        Force-push the image on digest mismatch, or if no digest is provided.
+    tls_verify : True
+        Verify TLS certificates when retrieving image metadata, passed to
+        `docker_registry.get_image_manifest` and
+        `docker_registry.push_docker_archive` (as `dest_tls_verify`).
```

Comment:
Same for the state module
you can extract the documentation with
```
salt-call sys.state_doc docker_registry.image_managed
```

Reply:
See prior comment.